lib/ansible/modules/cloud/azure/azure_rm_route.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_route
+version_added: "2.7"
+short_description: Manage Azure route resource.
+description:
+    - Create, update or delete a route.
+options:
+    resource_group:
+        description:
+            - name of resource group.
+        required: true
+    name:
+        description:
+            - name of the virtual network.
+        required: true
+    state:
+        description:
+            - Assert the state of the virtual network. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    address_prefix:
+        description:
+            - The destination CIDR to which the route applies.
+    next_hop_type:
+        description:
+            - The type of Azure hop the packet should be sent to.
+        choices:
+            - 'VirtualNetworkGateway'
+            - 'VnetLocal'
+            - 'Internet'
+            - 'VirtualAppliance'
+            - 'None'
+        default: 'None'
+    next_hop_ip_address:
+        description:
+            - The IP address packets should be forwarded to.
+            - Next hop values are only allowed in routes where the next hop type is VirtualAppliance.
+    route_table:
+        description:
+            - The name of the route table.
+        required: true
+
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a route
+      azure_rm_virtualnetwork:
+        name: foobar
+        resource_group: Testing
+        address_prefix: "10.1.0.0/16"
+        next_hop_type: "VirtualNetworkGateway"
```

Comment:
should be a need to quote this, we try and only quote values that need quoting to reduce the ambiguity for user's reading the documentation.